source/agora/test/Metadata.d

```diff
@@ -0,0 +1,69 @@
+/*******************************************************************************
+
+    Contains tests for the node network (P2P) behavior
+
+    Copyright:
+        Copyright (c) 2019 BOS Platform Foundation Korea
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+module agora.test.Metadata;
```

Comment:
Does it need its own test ? It should be more or less ignored in unittest for the time being.

Reply:
I'm confused. Are we doing TDD or not?

Reply:
We are but the point of abstracting away this is that it does nothing in unittests. So testing that it does nothing is a bit odd. In fact this test looks very much like the `Network` test.

Reply:
How do you make it do nothing in unittests then? You do one of these:

- Do a `version(unittest) { } else { this.metadata.loadFromDisk()` }`
- Subclass Metadata and then provide a File-based one by default, and an in-memory one in the unittests

Is there a third option? I really don't understand.